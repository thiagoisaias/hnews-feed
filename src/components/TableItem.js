import React from "react";
import styled from "styled-components";
import moment from "moment";
import PropTypes from "prop-types";

const Container = styled.div`
  border-bottom: 1px solid #eee;
  margin-bottom: 8px;
  padding-bottom: 6px;
`;

const MoreInfoContainer = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;
  font-size: 12px;
  max-width: 100%;
  white-space: nowrap;
  overflow: hidden;
`;

const MoreInfoItem = styled.li`
  display: block;
  border-left: 1px solid #e2e2e2;
  padding: 0 8px;
  cursor: default;
  color: #6f6f6f;

  &:first-child {
    padding-left: 0;
  }
  &:first-child {
    border: none;
  }
  @media (max-width: 425px) {
    display: ${props => (props.smallHide ? "none" : "list-item")};
  }
`;

const Title = styled.a`
  color: #484848;
  text-decoration: none;
`;

// const DismissButton = styled.button`
//   color: #484848;
//   border: none;
//   background-color: inherit;
//   outline: none;
//   padding: 0;
//   &:hover {
//     opacity: 0.7;
//     cursor: pointer;
//   }
// `;

const TableItem = props => {
  const { title, points, author, created_at, url } = props;
  return (
    <Container>
      <Title href={url} rel="noopener noreferrer" target="_blank">
        {title}
      </Title>
      <MoreInfoContainer>
        <MoreInfoItem> {points} points</MoreInfoItem>
        <MoreInfoItem> {author} </MoreInfoItem>
        <MoreInfoItem> {moment(created_at).fromNow()} </MoreInfoItem>
        <MoreInfoItem smallHide>
          <a href={url} rel="noopener noreferrer" target="_blank">
            ({url})
          </a>
        </MoreInfoItem>
      </MoreInfoContainer>
      {/* <DismissButton onClick={() => onDismiss(objectID)}>&times;</DismissButton> */}
    </Container>
  );
};

TableItem.propTypes = {
  // objectID: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  points: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  created_at: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  // onDismiss: PropTypes.func.isRequired
};

export default TableItem;
