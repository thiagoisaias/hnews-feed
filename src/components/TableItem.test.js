import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import TableItem from "./TableItem";
import moment from "moment";

describe("TableItem", () => {
  it("renders without crashing", () => {
    const itemTest = {
      objectID: 1,
      title: "random item test",
      points: 553.43,
      author: "Author Lorem",
      created_at: "2017-05-05T14:14:17.000Z",
      url: "random.url.test"
    }
    const TableItemForTest = <TableItem created_at={moment(itemTest.created_at).fromNow()} {...itemTest} />;
    const wrapper = shallow(TableItemForTest);
  });

  it("has valid snapshot", () => {
    const itemTest = {
      objectID: 1,
      title: "random item test",
      points: 553.43,
      author: "Author Lorem",
      created_at: "2017-05-05T14:14:17.000Z",
      url: "random.url.test"
    }
    const HeaderForTest = <TableItem created_at={moment(itemTest.created_at).fromNow()} {...itemTest} />;
    expect(shallowToJson(HeaderForTest)).toMatchSnapshot();
  });
});
