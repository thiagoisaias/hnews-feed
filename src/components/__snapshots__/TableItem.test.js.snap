// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableItem has valid snapshot 1`] = `null`;
